### ISSUE TITLE
Enquiry on 02_tutorial.ipynb

### ISSUE THREAD

**Issue by riven314** [rbracco/fastai2_audio#34]
I am new to this library and I am going through ```02_tutorial.ipynb```

In the notebook, when data bunch is constructed (i.e. ```dbunch = auds.dataloaders(p10speakers, bs = 64)```), I tried to display a batch by ```dbunch.show_batch()```. It displays a few snapshots of spectograms as shown:

![Capture](https://user-images.githubusercontent.com/21143399/79014711-0b214380-7b9e-11ea-8bc2-0c49f8d066ea.JPG)

However, I noticed that the resolution stated in the title (i.e. 128 x 251 px) doesn't align with the scale of xy-axis of the plot (i.e. 0-2 in x-axis, 0-8192 in y-axis). Is it an expected behavior?

Also, a batch has a shape of (64, 1, 128, 251). Does 128 refer to time axis or frequency axis?

**Comment by rbracco** [comment]
The spectrogram is a torch tensor and the dimensions given "128x251px" refer to both the shape of the image and the shape of the underlying tensor. The image is square because it is being stretched when displayed by either fastai2 or matplotlib (it's been a while since I've looked at this). Ideally it would not be displayed this way and would be shown as its actual shape.

The 0-2 is the time axis in seconds, but along that axis there are 251 pixels being displayed, one for every ~8ms. The 0-8192 is the frequency axis, represented by 128 pixels. Because there are more possible frequencies than pixels, what you're actually seeing is groups of frequencies being added together, and the sum of their energies being displayed as one pixel. For instance a pixel might be the sum of the energies in the range of 60hz-64hz. Since it is log scale these ranges get bigger the higher you get, and one of the final pixels might represent something like 9720hz-10000hz. 

Check out the intro to audio tutorial we did for fastai audio V1 here: https://github.com/mogwai/fastai_audio/tree/master/tutorials and let me know if you have any questions. Thanks!
